wire.md

```diff
@@ -0,0 +1,1008 @@
+* general enough to be used across different network transports
+* useful, even if written as a partial implementation
+* efficient in its use of network resources, by
+    * syncing only the relevant subsets of the full dataset, and
+    * being compact over the wire
+* not specific to any particular kind of database backend
+
+### 1.1 Terminology
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be
+interpreted as described in [BCP 14][BCP 14], [RFC 2119][RFC 2119], [RFC
+8174][RFC 8174] when, and only when, they appear in all capitals, as shown
+here.
+
+## 2. Scope
+This protocol focuses on the over-the-wire bytes that get sent between peers
+that enable the exchange of chat messages and user and channel information.
+
+This protocol does not specify encryption nor authentication of the connection,
+nor a mechanism for the discovery of network peers. These may be provided by
+other layers in userland (e.g. [Tor][Tor], [I2P][I2P]) or specified in future
+cable work.
+
+It is assumed that peers speaking cable are already authorized to access its
+data. Section [Security Considerations](#7-security-considerations) includes an
+analysis of the types of anticipated attacks that authorized members may still
+carry out.
+
+## 3. Definitions
+**client**: A running instance of a computer program that implements cable.
+
+**cabal**: An instance of a private group chat, whose client participants speak the cable protocol.
+
+**peer**: A client, to whom the referred to client is connected to over some transport protocol.
+
+**Ed25519**: A public-key cryptographic signature system.
+
+**hash**: A 32-byte BLAKE2b digest of a particular sequence of bytes.
+
+**public key**: An Ed25519 key that can be known by others.
+
+**private key**: An Ed25519 key, used for signing data. Kept secret to all but whoever controls it.
+
+**signature**: A unique sequence of binary data that can be produced by combining a private key and an input text. The signature can then be verified by the private key's corresponding public key.
+
+**user**: A pair of Ed25519 keys -- a public key and private key -- identifying a distinct person or program in a cabal. A user is known by others by their public key.
+
+**post**: A binary payload of a specific format, signed with the private key of the user who created it.
+
+**link**: A hash appearing in a post, which acts as a reference to another post.
+
+**channel**: A conceptual object with its own unique name, that users can participate in by authoring posts that reference the channel.
+
+**request**: A binary payload originating from a particular client.
+
+**response**: A binary payload, traversing the network from a particular client back to the client who issued the original request.
+
+**`req_id`**: A unique 32-bit number that uniquely identifies a request traversing in the network, and any corresponding responses.
```

Comment:
Fixed the duplicate "unique" -- thanks! You are technically correct that it refers to multiple messages: I'm thinking you're referring to the fact that since a peer forwarding a message would decrement its TTL, it's technically not the same message? I'll rephrase to say it identifies a "particular request".